### PR TITLE
[17.0][MIG] announcement: Migration to 17.0

### PR DESCRIPTION
**Pull request by chienandalu**
cc @Tecnativa TT50049

### CODE REVIEW

**Review by juanjosesegui-tecnativa — APPROVED** [review]

**Review by pedrobaeza — COMMENTED** [review]
Functionally is OK. About the formal aspect, you have to split the pre-commit auto-fixes in one commit, and the migration into another.

**Comment by chienandalu** [comment]
> you have to split the pre-commit auto-fixes in one commit

done

File: announcement/models/announcement.py [hunk 1]
```diff
@@ -96,7 +96,7 @@ def _inverse_specific_user_ids(self):
         itself"""
         for announcement in self:
             for user in announcement.specific_user_ids.filtered(
-                lambda x: announcement
+                lambda x, announcement: announcement
```
Comment (pedrobaeza) [inline announcement/models/announcement.py]: I think the second variable needs to be initialized

```suggestion
                lambda x, announcement=announcement: announcement
```

Comment (chienandalu) [reply announcement/models/announcement.py]: done

**Review by pedrobaeza — APPROVED** [review]
/ocabot merge nobump

**Pull request merged**